pkg/config/agent/agent.go

```diff
@@ -121,3 +121,36 @@ func Obfuscate(c Config) Config {
 	}
 	return c
 }
+
+// CPUs represents a list of CPUs.
+// It supports custom marshalling methods which allow
+// for specifying 'cpu' in 'scylla-manager-agent.yaml' config
+// as both a single int or an array of ints.
+type CPUs []int
+
+// UnmarshalYAML is a custom unmarshaller accepting both
+// single ints and int array.
+func (c *CPUs) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var v int
+	errV := unmarshal(&v)
+	if errV == nil {
+		*c = []int{v}
+		return nil
+	}
+	var arr []int
+	errArr := unmarshal(&arr)
+	if errArr == nil {
+		*c = arr
+		return nil
+	}
+	return errors.Wrap(multierr.Append(errV, errArr), "parse CPU as an int or int array")
+}
```

Comment:
What if there is no entry in config yaml ? 
It will produce a nil slice.

And that means agent will be not be pinned to any CPU, but only limited to one process.
It's expected, OK.